plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1723,7 +1726,9 @@ public HiveInsertTableHandle beginInsert(ConnectorSession session, ConnectorTabl
         if (isTransactional && retryMode != NO_RETRIES) {
```

Comment:
"create table as select" can work as  "insert into", so the table create in transaction will still be there after rollback...